test/runtests.jl

```diff
@@ -22,6 +22,11 @@ using Test
     @inferred a1(1.0)
 end
 
+@testset "fix(::Type, ...)" begin
+    f = @inferred fix(CartesianIndex, nothing, Some(1))
+    @test @inferred(f(2)) === CartesianIndex(2, 1)
+end
+
```

Comment:
I guess if a type occurs in the fixed arguments, inference will fail? We could add a broken test for that (not sure whats the best way to spell out a broken `@inferred` though.)

Reply:
Positional arguments are inferrable after https://github.com/JuliaLang/julia/pull/35980 (Julia >= 1.6) because the field type of `Some(::Type{T})` is now `Type{T}`.  If you want to fix it in earlier Julia, I think we can put the specialization for `Some(::Type)` in Compat.jl.

I think the only remaining problem is when a type is specified through a keyword argument.

Reply:
It's great that this works for `args` on 1.6. I think it's not worth fixing it for older versions, but it would be nice to have a broken test that reminds us about it. Especially the `kw` case, which is still open.

Reply:
Okay. I added a broken test for keyword argumets in c63b9e76652703abb51291aa4928751857b4a132.